### PR TITLE
added: add modal component

### PR DESCRIPTION
**Pull request by avriq**

### CODE REVIEW

File: components/modal/Modal.spec.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+  const modal = screen.queryByTestId('modal')
+  const text  = screen.queryByText('Modal Text')
+
+  expect(modal).toBeInTheDocument()
+  expect(text).toBeInTheDocument()
+
+  await fireEvent.keyDown(window, { key: 'Escape' })
+
+  expect(modal).toHaveAttribute('style', 'display: none;')
```
Comment (adenvt) [inline components/modal/Modal.spec.ts:82]: @avriq bisa pake `expect(modal).not.toBeVisible()` buat ngecek element hidden atau gak

**Review by adenvt — CHANGES_REQUESTED** [review]
@avriq revisi pack :')

File: components/modal/Modal.vue [hunk 2]
```diff
@@ -116,17 +116,16 @@ export default defineComponent({
...
-    function closeOnEsc (event: Event): void {
-      if (!props.noCloseOnEsc && event.key === 'Escape')
-        close(event)
+    function closeOnEsc (): void {
+      if (!props.noCloseOnEsc) {
+        onKeyStroke('Escape', (event) => {
+          close(event)
+        }, { eventName: 'keydown' })
+      }
     }
```
[the remark this answers is not in the document]
Comment (avriq) [reply components/modal/Modal.vue:125]: woalaa.. siuup :)))

File: components/overlay/Overlay.vue [hunk 3]
```diff
@@ -19,6 +19,7 @@ import Icon from './assets/icon-white.svg'
 export default defineComponent({
   setup () {
     const loading = useLoading({ elapsed: false })
+    // console.log(loading)
```
Comment (adenvt) [inline components/overlay/Overlay.vue:23]: delete aj pak @avriq

**Review by adenvt — APPROVED** [review]

**Pull request merged**